db-service/lib/common/DatabaseService.js

```diff
@@ -83,6 +99,7 @@ class DatabaseService extends cds.Service {
     if (!this.dbc) return
     else
       try {
+        await Promise.allSettled(this.activeQueries)
```

Comment:
Hanging queries is a possibility, but often caused by other side effects. Like the bug found in hdb [PR](https://github.com/SAP/node-hdb/pull/216#issue-1864835295).

An alternative solution would be to not release the database connection, but to disconnect/ destroy it in case of a rollback.

Reply:
> An alternative solution would be to not release the database connection, but to disconnect/ destroy it in case of a rollback.

i wouldn't be opposed to that, if the destroy doesn't await the stmt.exec. it's simple and effective, and rollbacks should happen very rarely, so very little impact.